_overviews/scala3-scaladoc/snippet-compiler.md

```diff
@@ -0,0 +1,223 @@
+---
+layout: multipage-overview
+title: Snippet checking
+partof: scala3-scaladoc
+num: 8
+previous-page: search-engine
+next-page: settings
+---
+
+The main functionality of documentation is to help people understand and use the project properly. Sometimes a part of a project needs few words to show its usage, but every developer knows that there are moments where description is not enough and nothing is better than a good ol’ example. A convenient way of providing examples in documentation is to create code snippets presenting usage of given functionality. The problem with code snippets is that simultaneously with project development, they need to be updated. Sometimes changes in one part of a project may break examples in other parts. The number of snippets and the amount of time passed since they’ve been written makes it impossible to remember every place where you need to fix them. After some time you realize that your documentation is a complete mess and you need to go through all examples and rewrite them. Many Scala 2 projects use typechecked markdown documentation with tut or mdoc. Almost everyone at least heard about these tools. As they turned out to be very useful and the Scala community successfully adopted them, we’re planning to incorporate the features of tut and mdoc into the compiler so that it’s included out of the box with Scaladoc.
+
+## Getting started
+
+By default, snippet validation is turned off for all snippets. It can be turned on by simply adding a following argument to Scaladoc:
+
+`-snippet-compiler:compile`
+
+For sbt, it might look like this:
+
+```scala
+Compile / doc / scalacOptions ++= Seq("-snippet-compiler:compile")
+```
+
+This option turns on snippet compiler for all `scala` snippets present in project (it recognizes them by ```scala part). Currently, snippet validation works in both docstrings written in Markdown and static sites.
+
+If you are starting new project, this configuration should be enough for you. However, in case you migrate existing project you might want to disable compilation for some snippets that can't be currently fixed.
+
+To do this, you can add a flag directly to the snippet:
+
+````
+```scala sc:nocompile
+// snippet
```

Comment:
```suggestion
// under the hood `map` is transformed into
List(1).map( _  + 1)(<implicits>)
```

Reply:
Or some other example that make sense not to compile

Reply:
changed